examples/field_examples/viking_graben_line12/rtm/rtm.jl

```diff
@@ -0,0 +1,188 @@
+mkpath(dir_out)
+
+# Load data and create data vector
+# block = segy_read(prestk_dir * prestk_file)
+container = segy_scan(prestk_dir, prestk_file, ["SourceX", "SourceY", "GroupX", "GroupY", "RecGroupElevation", "SourceSurfaceElevation", "dt"])
+d_obs = judiVector(container; segy_depth_key = segy_depth_key_rec)
+
+srcx = Float32.(get_header(container, "SourceX")[:,1])
+grpx = Float32.(get_header(container, "GroupX")[:,1])
+min_src_x = minimum(srcx)./1000f0
+max_src_x = maximum(srcx)./1000f0
+min_grp_x = minimum(grpx)./1000f0
+max_grp_x = maximum(grpx)./1000f0
+min_cdp_x = (min_src_x+min_grp_x)/2f0
+max_cdp_x = (max_src_x+max_grp_x)/2f0
+min_x = minimum([min_src_x, min_grp_x])
+max_x = maximum([max_src_x, max_grp_x])
+
+# helpful functions
+rho_from_slowness(m) = 0.23.*(sqrt.(1f0 ./ m).*1000f0).^0.25
+
+# Load starting model (mlog - slowness built with Vs from logs; mvsp - built from VSP)
+n, d, o, m0 = read(h5open(model_file, "r"), "n", "d", "o", "m")
+n = Tuple(Int64(i) for i in n)
+d = Tuple(Float32(i) for i in d)
+o = Tuple(Float32(i) for i in o)
+if modeling_type == "scalar"
+    model0 = Model(n, d, o, m0)
+elseif modeling_type == "acoustic"
+    rho0 = rho_from_slowness(m0)
+    model0 = Model(n, d, o, m0, rho=rho0)
+end
+
+x = (o[1]:d[1]:o[1]+(n[1]-1)*d[1])./1000f0
+z = (o[2]:d[2]:o[2]+(n[2]-1)*d[2])./1000f0
+
+global seabed_ind = Int.(round.(seabed./d[2]))
+if modeling_type == "scalar"
+    model0.m[:,1:seabed_ind] .= (1/vwater)^2
+elseif modeling_type == "acoustic"
+    model0.m[:,1:seabed_ind] .= (1/vwater)^2
+    model0.b[:,1:seabed_ind] .= 1f0/rhowater
+end
+
+# Set up wavelet and source vector
+src_geometry = Geometry(container; key = "source", segy_depth_key = segy_depth_key_src)
+
+# sampling frequency
+fs = 1000f0/src_geometry.dt[1]
+# Nyquist frequency
+f_nyq = fs/2f0
+
+# setup wavelet
+wavelet_raw = readdlm(wavelet_file, skipstart=wavelet_skip_start)
+itp = LinearInterpolation(0:wavelet_dt:wavelet_dt*(length(wavelet_raw)-1), wavelet_raw[:,1], extrapolation_bc=0f0)
+wavelet = Matrix{Float32}(undef,src_geometry.nt[1],1)
+wavelet[:,1] = itp(0:src_geometry.dt[1]:src_geometry.t[1])
+q = judiVector(src_geometry, wavelet)
+
+############################################## FWI #################################################
+
+grad_mem = 40 # Based on n and CFL condition
+
+mem = Sys.free_memory()/(1024^3)
+t_sub = max(1, ceil(Int, nworkers()*grad_mem/mem))
+@info "modeling_type: $modeling_type"
+@info "subsampling factor: $t_sub"
+
+# JUDI options
+jopt = JUDI.Options(
+    space_order=16,
+    limit_m = true,
+    buffer_size = buffer_size,
+    optimal_checkpointing=false,
+    IC = "isic",
+    # subsampling_factor=t_sub     # subsampling_factor with space_order set leads to exception (probably bug)
+    )
+
+# Right-hand preconditioners (model topmute)
+idx_wb = find_water_bottom(reshape(model0.m, model0.n))
+Tm = judiTopmute(model0.n, idx_wb, 10)  # Mute water column
+S = judiDepthScaling(model0)
```

Comment:
There is a `judiIllumination` now in JUDI that might have been slightly better

Reply:
Don't how to use it yet. But interested in it.